pages/k8s/upgrading.md

```diff
@@ -0,0 +1,319 @@
+---
+title: "Upgrading | Canonical Distribution of Kubernetes&reg;"
+keywords: juju, upgrading, track, version
+tags: [operating]
+sidebar: k8smain-sidebar
+permalink: upgrading.html
+layout: base
+toc: False
+summary: This page is currently a work in progress. For existing documentation, please visit <a href="https://kubernetes.io/docs/getting-started-guides/ubuntu/"> https://kubernetes.io/docs/getting-started-guides/ubuntu/ </a>
+---
+
+# Upgrading
+
+It is recommended that you keep your **Kubernetes** deployment updated to the
+latest available stable version. You should also update the other applications
+which make up the **Canonical Distribution of Kubernetes<sup>&reg;</sup>.**
+Keeping up to date makes sure you have the latest bug-fixes and security
+patches for smooth operation of your cluster.
+
+There is no set cadence to the releases of **Kubernetes**,  but you can check
+the latest release version on the  
+[Kubernetes release page on GitHub][k8s-release].  The **CDK** is kept in close sync
+with upstream Kubernetes.
+
+<div class="p-notification--information">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Note:</span>
+<strong>Kubernetes</strong> will automatically handle patch releases. This means that
+the cluster will perform an unattended automatic upgrade between patch versions, e.g.
+1.10.7 to 1.10.8. Attended upgrades are only required when you wish to upgrade a minor
+version, e.g. 1.9.x to 1.10.x.
+  </p>
+</div>
+
+You can see which version of each application is currently deployed by running
+
+```bash
+juju status
+```
+
+The 'App' section of the output lists each application and its version number.
+Note that this is the version of the upstream application deployed. The version
+of the Juju charm is indicated under the column titled 'Rev'. The charms may be
+updated in between new versions of the application.
+
+## Before you begin
+
+As with all upgrades, there is a possibility that there may be unforeseen
+difficulties. It is **highly recommended that you make a backup** of any
+important data, including any running workloads. For more details on creating
+backups, see the separate [documentation on backups][backups].
+
+You should also make sure:
+
+  * The machine from which you will perform the backup has sufficient internet access to
+      retrieve updated software
+  * Your cluster is running normally
+  * You read the [Upgrade notes][notes] to see if any caveats apply to the versions you are
+     upgrading to/from
+  * You read the [Release notes][release-notes] for the version you are upgrading to,
+      which will alert you to any important changes to the operation of your cluster
+
+## Infrastructure updates
+
+The applications which run alongside the core Kubernetes components can be
```

Comment:
The instructions sort of imply that it doesn't matter if you are using a much newer version of, for example, easyrsa.  Maybe it doesn't, but I guess theoretically, you are not then running a particular version of CDK. I don't know if this is likely to cause issues.

Reply:
The reality is we are usually the ones updating these charms, so it's unlikely to cause problems. 

But, when we publish a new bundle, each component is pinned to a particular revision - the latest one that we recommend and have tested. It would be nice to convey those recommendations for those upgrading existing clusters. Something to keep in mind for the future - maybe open a new issue about that so we don't forget?

Reply:
yup - 
https://github.com/juju-solutions/kubernetes-docs/issues/21